src/cmd/serve.rs

```diff
@@ -858,4 +863,194 @@ mod tests {
+        let output_dir: Option<PathBuf> = None;
+        let base_url: Option<String> = None;
+        let no_port_append = false;
+        let ws_port: Option<u16> = None;
+        let expected_base_url = String::from("http://127.0.0.1:1111");
+
+        create_and_verify_new_site(
+            interface,
+            interface_port,
+            output_dir.as_deref(),
+            base_url.as_deref(),
+            no_port_append,
+            ws_port,
+            expected_base_url,
+        );
+    }
+
+    #[test]
+    fn test_create_new_site_without_protocol_with_port_with_mounted_path() {
+        let interface = IpAddr::from_str("127.0.0.1").unwrap();
+        let interface_port = 1111;
+        let output_dir: Option<PathBuf> = None;
+        let base_url: Option<String> = Some(String::from("localhost/path/to/site"));
+        let no_port_append = false;
+        let ws_port: Option<u16> = None;
+        let expected_base_url = String::from("http://localhost:1111/path/to/site");
+
+        create_and_verify_new_site(
+            interface,
+            interface_port,
+            output_dir.as_deref(),
+            base_url.as_deref(),
+            no_port_append,
+            ws_port,
+            expected_base_url,
+        );
+    }
+
+    #[test]
+    fn test_create_new_site_without_protocol_without_port_without_mounted_path() {
+        let interface = IpAddr::from_str("127.0.0.1").unwrap();
+        let interface_port = 1111;
+        let output_dir: Option<PathBuf> = None;
+        let base_url: Option<String> = Some(String::from("example.com"));
+        let no_port_append = true;
+        let ws_port: Option<u16> = None;
+        let expected_base_url = String::from("http://example.com");
+
+        // Note that no_port_append only works if we define a base_url
+
+        create_and_verify_new_site(
+            interface,
+            interface_port,
+            output_dir.as_deref(),
+            base_url.as_deref(),
+            no_port_append,
+            ws_port,
+            expected_base_url,
+        );
+    }
+
+    #[test]
+    fn test_create_new_site_with_protocol_without_port_without_mounted_path() {
+        let interface = IpAddr::from_str("127.0.0.1").unwrap();
+        let interface_port = 1111;
+        let output_dir: Option<PathBuf> = None;
+        let base_url: Option<String> = Some(String::from("https://example.com"));
+        let no_port_append = true;
+        let ws_port: Option<u16> = None;
+        let expected_base_url = String::from("https://example.com");
+
+        // Note that no_port_append only works if we define a base_url
+
+        create_and_verify_new_site(
+            interface,
+            interface_port,
+            output_dir.as_deref(),
+            base_url.as_deref(),
+            no_port_append,
+            ws_port,
+            expected_base_url,
+        );
+    }
+
+    #[test]
+    fn test_create_new_site_with_protocol_without_port_with_mounted_path() {
```

Comment:
Looks like this fails on windows?

Reply:
All the tests were green prior to rebase. `The process cannot access the file because it is being used by another process.` This seems like a concurrency issue with the test runner.

Reply:
Can we try re-running and see if it was transient?

Reply:
looks good!

Reply:
It does fail very often on CI in the end, I'll disable it for windows

Reply:
Well that's unfortunate. I have a Windows VM I'll see if I can reproduce the failure outside of CI.

Reply:
It is peculiar if it only fails on the one test you disabled as there are several that are virtually identical, just using slightly different parameters.

I expect the failure to reemerge on the next active test of that type...

Reply:
Yeah it's weird. I've disabled it so i can release 0.19 but it would be interesting to get some investigation done